scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -748,7 +757,99 @@ private void deleteActiveSchedules(String appId) {
 			throw new SchedulerInternalException("Database error", dve);
 		}
 	}
-
+	
+	@Transactional
+	public SynchronizeResult synchronizeSchedules(){
+		int createCount = 0;
+		int updateCount = 0;
+		int deleteCount = 0;
+		Map<String, PolicyJsonEntity> policyMap = new HashMap<String, PolicyJsonEntity>();
+		Map<String, String> appIdAndGuidMap = new HashMap<String, String>();
+		List<PolicyJsonEntity> policyList = policyJsonDao.getAllPolicies();
+		List recurringScheduleList = recurringScheduleDao.getDistinctAppIdAndGuidList();
+		List specificDateScheduleList = specificDateScheduleDao.getDistinctAppIdAndGuidList();
+		//create or updated
+		if(policyList.size() > 0){
+			for(PolicyJsonEntity policy : policyList){
+				policyMap.put(policy.getAppId(), policy);
+			}
+		}
+		
+		if(recurringScheduleList.size() > 0){
+			for(Object ro : recurringScheduleList){
+				Object[] roArray = (Object[])ro;
+				appIdAndGuidMap.put((String)(roArray[0]), (String)(roArray[1]));
+			}
+		}
+		
+		if(specificDateScheduleList.size() > 0){
+			for(Object so : specificDateScheduleList){
+				Object[] soArray = (Object[])so;
+				appIdAndGuidMap.put((String)(soArray[0]), (String)(soArray[1]));
+			}
+		}
+		
+		List<PolicyJsonEntity> toCreatePolicyList = new ArrayList<PolicyJsonEntity>();
+		Set<String> toDeletedAppIds = new HashSet<String>();
+		for (String appIdInPolicy : policyMap.keySet()) {
+			if (policyMap.get(appIdInPolicy).getSchedules() != null
+					&& policyMap.get(appIdInPolicy).getSchedules().getSchedules() != null) {
+				if (policyMap.get(appIdInPolicy).getSchedules().getSchedules().getRecurringSchedule() != null
+						&& policyMap.get(appIdInPolicy).getSchedules().getSchedules().getRecurringSchedule()
+								.size() > 0) {
```

Comment:
Since this piece of code was part of this PR so made this comment. Decision on that can be made if there is a consensus. Your thoughts on this @boyang9527, @paltanmoy